File: api/src/types/magic_school.rs
#[derive(Debug, Clone, Copy, PartialEq, sqlx::Type, serde::Serialize, serde::Deserialize)]
#[sqlx(type_name = "magic_school", rename_all = "snake_case")]
#[serde(rename_all = "snake_case")]
pub enum MagicSchool {
    Abjuration,
    Conjuration,
    Divination,
    Enchantment,
    Evocation,
    Illusion,
    Necromancy,
    Transmutation,
}

File: view/src/lib/types/MagicSchool.ts
export enum MagicSchool {
  Abjuration = "abjuration",
  Conjuration = "conjuration",
  Divination = "divination",
  Enchantment = "enchantment",
  Evocation = "evocation",
  Illusion = "illusion",
  Necromancy = "necromancy",
  Transmutation = "transmutation",
}

export const DisplayMagicSchool: Readonly<Record<MagicSchool, string>> = {
  [MagicSchool.Abjuration]: "Abjuration",
  [MagicSchool.Conjuration]: "Conjuration",
  [MagicSchool.Divination]: "Divination",
  [MagicSchool.Enchantment]: "Enchantment",
  [MagicSchool.Evocation]: "Evocation",
  [MagicSchool.Illusion]: "Illusion",
  [MagicSchool.Necromancy]: "Necromancy",
  [MagicSchool.Transmutation]: "Transmutation",
} as const;

export const MagicSchools: MagicSchool[] = Object.values(MagicSchool);
